src/components/File.jsx

```diff
@@ -0,0 +1,78 @@
+import React, { Component } from 'react'
+import classNames from 'classnames'
+import { Link } from 'react-router'
+
+import styles from '../styles/table'
+import { translate } from '../lib/I18n'
+import FilenameInput from '../components/FilenameInput'
+
+const splitFilename = filename => {
+  let dotIdx = filename.lastIndexOf('.') - 1 >>> 0
```

Comment:
I don't like that kind of trick either, but the alternative is rather verbose and miss some edge cases, so...